src/main/scala/edu/ie3/simona/model/thermal/ThermalHouse.scala

```diff
@@ -113,24 +136,176 @@ final case class ThermalHouse(
+        val hoursDiff = (timeDiffSeconds / 3600).toInt
+
+        Some(
+          (1 to hoursDiff + 1).map(hour => (lastStateTime.getHour + hour) % 24)
+        )
+
+      } else if simulationTime.getHour != lastStateTime.getHour then {
+        Some(Seq(simulationTime.getHour))
+      } else {
+        None
+      }
+    }
+  }
+
+  /** Calculate the energy demand for warm water at the instance in question.
     *
-    * In edge cases, i.e. within the tolerance margin of target temperatures,
-    * the temperature difference can be negative. For these cases we set the
-    * temperature difference to zero, resulting in an energy demand of 0 kWh.
+    * @param hoursWaterDemandToDetermine
+    *   Sequence of hours for which the energy demand should be determined.
+    * @return
+    *   The needed energy for hot domestic water for the questioned hours.
+    */
+
+  def energyDemandDomesticHotWater(
+      hoursWaterDemandToDetermine: Option[Seq[Int]]
+  ): ThermalEnergyDemand = {
+    val totalEnergyDemand = hoursWaterDemandToDetermine match {
+      case Some(hours) =>
+        hours.foldLeft(zeroKWh) { (accumulator, hour) =>
+          accumulator + calculateThermalEnergyOfWaterDemand(hour)
+        }
+      case None =>
+        zeroKWh
+    }
+    ThermalEnergyDemand(totalEnergyDemand, totalEnergyDemand)
+  }
+
+  /** Calculate the energy demand for warm water for some specific hour at the
+    * instance in question.
+    *
+    * @param hour
+    *   The hour for that the energy demand should be determined.
+    * @return
+    *   The needed energy for domestic hot water in the questioned hour.
+    */
+
+  private def calculateThermalEnergyOfWaterDemand(hour: Int): Energy = {
+    val waterDemand =
+      waterDemandOfHour(hour, houseInhabitants, housingType)
+    thermalEnergyDemandWater(
+      waterDemand,
+      lowerTemperatureTapWater,
+      upperTemperatureTapWater,
+    )
+  }
+
+  /** Calculate the energy required to heat up a given volume of water from a
+    * start to an end temperature.
     *
+    * @param waterDemand
+    *   Water volume to get heated up.
     * @param startTemperature
-    *   The starting temperature.
+    *   Starting Temperature.
+    * @param endTemperature
+    *   End Temperature.
+    * @return
+    *   The needed energy.
+    */
+
+  private def thermalEnergyDemandWater(
+      waterDemand: Volume,
+      startTemperature: Temperature,
+      endTemperature: Temperature,
+  ): Energy = {
+    if endTemperature < startTemperature then
+      throw new RuntimeException(
+        s"End temperature of $endTemperature is lower than the start temperature $startTemperature for the water heating system."
+      )
+
+    val specificHeatDemandWater = KilowattHoursPerKelvinCubicMeters(
+      Joules(4184e3).toKilowattHours
+    )
+    val temperatureDelta = endTemperature - startTemperature
+
+    specificHeatDemandWater.calcEnergyToHeat(waterDemand, temperatureDelta)
```

Comment:
There is already a method (`calcEnergy`) that takes a volume and two temperatures. This method also takes care of calculating the temperature difference.

Reply:
good point...